ledger/controllers/grants.js

```diff
@@ -368,37 +488,77 @@ v1.claimGrant = { handler: (runtime) => {
 
     return reply(result)
   }
-},
-  description: 'Request a grant for a wallet',
-  tags: [ 'api' ],
+}
 
-  plugins: {
-    rateLimit: {
-      enabled: rateLimitEnabled,
-      rate: (request) => claimRate
+async function safetynetCheck (debug, runtime, request, promotion, wallet) {
+  const {
+    config,
+    database
+  } = runtime
+  const {
+    captcha
+  } = config
+  const {
+    headers
+  } = request
+  const {
+    paymentId
+  } = wallet
+  const url = `${captcha.url}/v1/attestations/safetynet`
+  const captchaHeaders = {
+    'Authorization': 'Bearer ' + captcha.access_token,
+    'Content-Type': 'application/json'
+  }
+  const wallets = database.get('wallets', debug)
+  const body = JSON.stringify({
+    token: headers['safetynet-token']
+  })
+
+  const payload = await braveHapi.wreck.post(url, {
```

Comment:
what error do we return if this fails? 5xx or 4xx?

it should be 5xx if there was a 5xx or timeout and 4xx if the response was a 4xx

Reply:
throw ends up being the same thing as reply. statusCode is passed through via throw